src/app/auth/auth.component.html

```diff
@@ -29,7 +29,12 @@
         </tr>
       </table>
       <mat-card-actions>
-        <button type="submit" [disabled]="loginForm.invalid" mat-stroked-button color="primary">Sign In</button>
+        <button type="submit" [disabled]="loginForm.invalid" mat-stroked-button color="primary">
+          <div>
+            <span *ngIf="this.authState.toString() === '0'">Sign In</span>
+            <mat-spinner [diameter]="30" *ngIf="this.authState.toString() !== '0'"></mat-spinner>
```

Comment:
Can abstract these out into a function in `auth.component.ts`? We can use the AuthState Enum instead of 0.

Reply:
Sure! 👍